### PR TITLE
Tests for quickestOrCheapest

### PR DESCRIPTION
**Pull request by Dyktus**

### CODE REVIEW

File: test/journey/filter/QuickestOrCheapest.spec.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            none,
+            0,
+            false,
+        ),
+    ),
+];
+
+const fareOptionFactories = [
+    getFareOptionWithIdFactory(
+        (fs, px, av): Option<FareOption> => {
```
Comment (orzeuek) [inline test/journey/filter/QuickestOrCheapest.spec.ts]: can you decrypt `fs, px, av` ?

File: test/journey/filter/QuickestOrCheapest.spec.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            publicRailcard,
+            none,
+            none,
+            0,
+            false,
+        ),
+    ),
+];
+
+const fareOptionFactories = [
```
Comment (orzeuek) [inline test/journey/filter/QuickestOrCheapest.spec.ts]: why is that array ? 

only difference between elements is different journeyFare - can't it be a parameter or something ?

File: test/journey/filter/QuickestOrCheapest.spec.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+};
+
+describe('QuickestOrCheapest', () => {
+
+    it('quickestOrCheapest function filter journeys properly', async () => {
+        const out = [quickestOrCheapest].reduce((journeys, filter) => filter(fareCache, journeys), outwardJourneys);
+
+        chai.expect(JSON.parse(JSON.stringify(out[0]))).to.deep.equal(
+            {
+                "legs": [{
```
Comment (orzeuek) [inline test/journey/filter/QuickestOrCheapest.spec.ts]: shouldn't it be a full journey object not just a list of legs ?

Comment (Dyktus) [reply test/journey/filter/QuickestOrCheapest.spec.ts]: It's not full Journey object as in here https://github.com/assertis/odyssey-service/blob/master/test/views/JourneyViewFactory.spec.ts#L15
But it is same journey part which is also returned here 
https://github.com/assertis/odyssey-service/blob/master/src/journey/JourneyPlanner.ts#L61

It's minimum just legs and departureDate
https://github.com/open-track/journey-planner/blob/master/src/journey/Journey.ts

File: test/journey/filter/QuickestOrCheapest.spec.ts (hunk 2 above)
Comment (Dyktus) [reply test/journey/filter/QuickestOrCheapest.spec.ts]: I've changed it to use parameter.

**Pull request merged**